backend/elasticManager.js

```diff
@@ -0,0 +1,236 @@
+import elasticsearch from 'elasticsearch';
+import {merge} from 'ramda';
+import YAML from 'yamljs';
+
+import {parseES} from './parse';
+import {ARGS} from './args';
+import {APP_NOT_CONNECTED, AUTHENTICATION} from './errors';
+
+const EMPTY_TABLE = {
+    columnnames: ['NA'],
+    rows: [['empty table']],
+    ncols: 1,
+    nrows: 1
+};
+
+const isEmpty = (table) => {
+    return table.length === 0;
+};
+
+const assembleTablesPreviewMessage = (tablePreviews) => {
+
+    /*
+        topRows is an array of one or many responses of top 5 rows queries
+        [ {'table1':[top5rows]}, {'table2': [top5rows]} ...]
+    */
+
+    let parsedRows;
+
+    return tablePreviews.map( (tablePreview) => {
+        const tableName = Object.keys(tablePreview);
+        const rawData = tablePreview[tableName];
+        parsedRows = (isEmpty(rawData)) ? EMPTY_TABLE : parseES(rawData);
+        return {[tableName]: parsedRows};
+    });
+
+};
+
+export class ElasticManager {
+
+    constructor(Logger) {
+        this.log = Logger.log;
+        this.raiseError = Logger.raiseError;
+        this.sessionSelected = 0;
+        this.sessions = {};
+    }
+
+    setSessionSelected(sessionSelected) {
+        this.sessionSelected = sessionSelected;
+    }
+
+    getDialect() {
+        return this.sessions[this.sessionSelected].options.dialect;
+    }
+
+    getConnection(responseSender) {
+        return () => responseSender({error: null});
+    }
+
+
+    createConnection(configuration) {
+        const {host, port, username, password} = configuration;
+        this.log(`Creating a connection to elasticsearch at ${host}:${port}`, 1);
+        this.sessions[this.sessionSelected] = new elasticsearch.Client({
+          host: `${host}:${port}`,
+          auth: `${username}:${password}`
+        });
+    }
+
+    authenticate(responseSender) {
+        this.log('Authenticating connection.');
+        // when already logged in and simply want to check connection
+        if (!this.sessions[this.sessionSelected]) {
+			this.raiseError(
+                merge(
+                    {message: APP_NOT_CONNECTED},
+                    {name: 'ConnectionError'}
+                ),
+                responseSender
+			);
+		} else {
+            return new Promise((resolve, reject) => {
+                this.sessions[this.sessionSelected].ping({
+                    // ping usually has a 3000ms timeout
+                    requestTimeout: Infinity,
+                    // undocumented params are appended to the query string
+                    hello: 'hello elasticsearch!'
```

Comment:
is there supposed to be a username and password here? if we're not supporting username and password, can we remove it from the UI?